backend/user-service/user-api.yaml

```diff
@@ -63,6 +71,7 @@ components:
       name: role
       in: query
       required: true
+      allowEmptyValue: true
```

Comment:
I'm confused what this configuration means. It's required, but an empty value is allowed? What does it mean for it to be empty?

Reply:
Need to allow empty as the empty string is an acceptable value. It would be when the user first sign in but never assigned a role yet. The `required` is not as important I think. But since it works, I left it there. I think it means `/user/xxx?role=` will work, but `/user/xxx` won't. Haven't look too deep into how does the generated code parse it. It works for our use case.

Reply:
ok, thanks for explaining.